### PR TITLE
Peer Message Receiver Refactor

### PR DESCRIPTION
**Pull request by benthecarman**
Refactors `PeerMessageReceiver` to have it's parent `Node` instead of a `DataMessageHandler` and then give the `DataMessageHandler` to the `Node` this will allow the `Node` to have different peers that are all contributing and looking at the same view of the blockchain

### CODE REVIEW

File: node-test/src/test/scala/org/bitcoins/node/networking/peer/PeerMessageHandlerTest.scala [hunk 1]
```diff
@@ -39,35 +39,44 @@ class PeerMessageHandlerTest
   implicit protected lazy val chainConfig: ChainAppConfig =
     cachedConfig.chainConf
 
-  behavior of "PeerHandler"
+  override def beforeAll(): Unit = {
+    implicit val chainConf: ChainAppConfig = cachedConfig.chainConf
```
Comment (Christewart) [inline node-test/src/test/scala/org/bitcoins/node/networking/peer/PeerMessageHandlerTest.scala]: This is probably going to need to change as per work on #2943 , we probably don't want to block this work on that though.

File: node/src/main/scala/org/bitcoins/node/Node.scala [hunk 2]
```diff
@@ -45,12 +45,14 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   val peer: Peer
 
-  protected val initialSyncDone: Option[Promise[Done]]
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/Node.scala:48]: So this is totally gone now? How do we know when IBD is done for a specific node ?

File: node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageSender.scala [hunk 3]
```diff
@@ -72,7 +72,10 @@ case class PeerMessageSender(client: P2PClient)(implicit conf: NodeAppConfig)
 
   def sendVersionMessage(chainApi: ChainApi)(implicit
       ec: ExecutionContext): Future[Unit] = {
-    chainApi.getBestHashBlockHeight().flatMap { height =>
+    val heightF =
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageSender.scala]: Hmm... what is this for ? Why is this needed now?

**Review by Christewart — COMMENTED** [review]
Related to #1764

File: node/src/main/scala/org/bitcoins/node/Node.scala (hunk 2 above)
Comment (benthecarman) [reply node/src/main/scala/org/bitcoins/node/Node.scala:48]: We never used it anywhere, we just use the `syncing` value

File: node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageSender.scala (hunk 3 above)
Comment (benthecarman) [reply node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageSender.scala]: Ran into this with some of the testing stuff, basically, if our db is empty then we should return a `0`

Comment (Christewart) [reply node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageSender.scala]: Ok here is a patch to fix this so we can remove this: 

https://github.com/Christewart/bitcoin-s-core/commit/69fa407435501bd0f1e4cf8d585c684f37608a0b

File: node/src/main/scala/org/bitcoins/node/Node.scala [hunk 4]
```diff
@@ -45,12 +45,14 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
...
+  def dataMessageHandler: DataMessageHandler
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/Node.scala:54]: /** The current data message handler.
    * It should be noted that the dataMessageHandler contains
    * chainstate. When we update with a new chainstate, we need to
    * maek sure we update the [[DataMessageHandler]] via [[updateDataMessageHandler()]]
    * to make sure we don't corrupt our chainstate cache */

File: node/src/main/scala/org/bitcoins/node/Node.scala [hunk 5]
```diff
@@ -206,7 +198,7 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
       // Get all of our cached headers in case of a reorg
       val cachedHeaders =
         blockchains.flatMap(_.headers).map(_.hashBE.flip)
-      peerMsgSenderF.map(_.sendGetHeadersMessage(cachedHeaders))
+      peerMsgSender.sendGetHeadersMessage(cachedHeaders)
```
Comment (Christewart) [inline node/src/main/scala/org/bitcoins/node/Node.scala]: This returns a `Future`, can we move this up into the for comprehension? I.e. 

```scala
// Get all of our cached headers in case of a reorg
cachedHeaders = blockchains.flatMap(_.headers).map(_.hashBE.flip)
_ <- peerMsgSender.sendGetHeadersMessage(cachedHeaders)
 ```

**Review by Christewart — CHANGES_REQUESTED** [review]

**Review by Christewart — APPROVED** [review]

**Review by nkohen — APPROVED** [review]
